src/lib/Webhook.coffee

```diff
@@ -0,0 +1,61 @@
+request = require "requestretry"
+
+DEFAULT_TIMEOUT = 10 * 1000
+DEFAULT_MAX_ATTEMPTS = 3
+
+class Webhook
+
+  constructor: (options = {}) ->
+    @options = options
+    @timeout = DEFAULT_TIMEOUT
+    @maxAttempts = DEFAULT_MAX_ATTEMPTS
+
+  @detectEmoji: (emoji) ->
+    obj = {}
+
+    unless emoji
+      obj.key = "icon_emoji"
+      obj.val = ""
+      return obj
+
+    obj.key = if emoji.match(/^http/) then "icon_url" else "icon_emoji"
+    obj.val = emoji
+    return obj
+
+  respond: (options, callback) ->
+
+    if typeof options is "object"
+      text = options.text
+    else
+      text = options
+      options = {}
+
+    emoji = Webhook.detectEmoji(options.icon_emoji or @options.response_type)
+
+    payload =
+      response_type: options.response_type or @options.response_type or 'ephemeral'
```

Comment:
why do we check options.response_type here, but not on line 33? also how could that represent ephemeral and potentially an emoji value?